frame/election-provider-support/src/lib.rs

```diff
@@ -333,6 +333,32 @@ pub trait SortedListProvider<AccountId> {
 	fn clear();
 	/// Sanity check internal state of list. Only meant for debug compilation.
 	fn sanity_check() -> Result<(), &'static str>;
+
+	/// `id` is in a position in the list that corresponds to `weight`
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn is_in_pos(_id: &AccountId, _weight: VoteWeight, mock: bool) -> bool {
+		mock
+	}
+
+	/// Wether the voter is the head of the notional bag for the given weight. Only relevant for
+	/// implementations that use bags.
+	///
+	/// `mock` parameter is for implementations where this method is not relevant.
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn is_bag_head(_: &AccountId, _: VoteWeight, mock: bool) -> bool {
+		mock
+	}
+
+	/// If `who` changes by the returned amount they are guaranteed to have a worst case change
+	/// in their list position.
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn weight_update_worst_case(_who: &AccountId, is_increase: bool) -> VoteWeight {
+		if is_increase {
+			VoteWeight::MAX
+		} else {
+			1
+		}
+	}
```

Comment:
```suggestion
	/// `id` is in a position in the list that corresponds to `weight`
	#[cfg(any(feature = "runtime-benchmarks", test))]
	fn is_in_pos(_id: &AccountId, _weight: VoteWeight, mock: bool) -> bool;

	/// Wether the voter is the head of the notional bag for the given weight. Only relevant for
	/// implementations that use bags.
	///
	/// `mock` parameter is for implementations where this method is not relevant.
	#[cfg(any(feature = "runtime-benchmarks", test))]
	fn is_bag_head(_: &AccountId, _: VoteWeight, mock: bool) -> bool;

	/// If `who` changes by the returned amount they are guaranteed to have a worst case change
	/// in their list position.
	#[cfg(any(feature = "runtime-benchmarks", test))]
	fn weight_update_worst_case(_who: &AccountId, is_increase: bool) -> VoteWeight;
```

Reply:
I don't get why we need the mock?

Reply:
if you are getting complaints that this method is not implemented when running `cargo test`, it is because: 

1. here you state `#[cfg(any(feature = "runtime-benchmarks", test))]` but in the implementation you say `#[cfg(feature = "runtime-benchmarks")]`. 
2. Overall, I learned that while this `any(feature = "runtime-benchmarks", test)` has one small benefit (you run benchmark tests as well when you run `cargo test`), it is a hornet's nest and I moved away from it. No longer needed to have `any(feature = "runtime-benchmarks", test)` on anything, including the `benchmarks.rs` and `mock.rs`, with only flag `cfg(feature = "runtime-benchmarks")`. If you want to run the benchmark tests, adding `--features runtime-benchmarks` is easy enough.